### PR TITLE
Fix null pointer exception

### PR DESCRIPTION
**Pull request by emswbandara**
solves issue-2643

### CODE REVIEW

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 1]
```diff
@@ -949,9 +951,11 @@ private void handleLogoutResponseFromFramework(HttpServletRequest request,
             if (StringUtils.isBlank(acsUrl) && sessionDTO.getIssuer() != null) {
                 SAMLSSOServiceProviderDO serviceProviderDO = getSPConfig(SAMLSSOUtil.getTenantDomainFromThreadLocal(),
                         sessionDTO.getIssuer());
-                acsUrl = serviceProviderDO.getSloResponseURL();
-                if (StringUtils.isBlank(acsUrl)) {
-                    acsUrl = serviceProviderDO.getDefaultAssertionConsumerUrl();
+                if (serviceProviderDO != null) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: Shouldnt we break the flow if SP is null?

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 2]
```diff
@@ -418,18 +418,20 @@ private void handleIdPInitSSO(HttpServletRequest req, HttpServletResponse resp,
                             SAMLSSOServiceProviderDO serviceProviderDO = getSPConfig(SAMLSSOUtil
                                             .getTenantDomainFromThreadLocal(),
                                     SAMLSSOUtil.splitAppendedTenantDomain(issuer));
-                            //for IDP init SLO, priority should be given to SLO response URL over default ACS.
-                            acsUrl = serviceProviderDO.getSloResponseURL();
-                            if (StringUtils.isBlank(acsUrl)) {
-                                acsUrl = serviceProviderDO.getDefaultAssertionConsumerUrl();
-                            }
+                            if (serviceProviderDO != null) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: What happens if ServiceProviderDO is null?
Do we throw an exception and stop the flow?

Comment (emswbandara) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: here acsUrl is required for an error scenario path. serviceProviderDO is obtained to set the acsUrl as a query param to saml notification.do page if it is null. This shouldn't break the flow.

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java (hunk 1 above)
Comment (emswbandara) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: here acsUrl is required for error scenario paths. serviceProviderDO is obtained to set the acsUrl as a query param to saml notification.do page if it is null. This shouldn't break the flow.

**Pull request merged**